packages/react-aria-components/stories/ListBox.stories.tsx

```diff
@@ -392,7 +393,48 @@ export function VirtualizedListBoxGrid({minSize, maxSize}) {
   );
 }
 
-VirtualizedListBoxGrid.args = {
-  minSize: 80,
-  maxSize: 100
+export const VirtualizedListBoxGrid = {
+  render(args) {
+    return <VirtualizedListBoxGridExample {...args} />;
+  },
+  args: {
+    minSize: 80,
+    maxSize: 100,
+    preserveAspectRatio: false
+  }
 };
+
+let lorem = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.'.split(' ');
+
+export function VirtualizedListBoxWaterfall({minSize = 80, maxSize = 100}) {
+  let items: {id: number, name: string}[] = [];
+  for (let i = 0; i < 1000; i++) {
+    let words = Math.max(2, Math.floor(Math.random() * 25));
+    let name = lorem.slice(0, words).join(' ');
+    items.push({id: i, name});
+  }
+
+  let layout = useMemo(() => new WaterfallLayout(), []);
```

Comment:
is there any legitimate case for invalidating this and creating a new instance? if not, then should people just pass the layout class into the layout prop?
could use layoutOptions for any initial/setup values they wanted

I assume we'd want to discourage people from sending values to the constructor, instead making use of the layoutOptions

Reply:
hmm yeah maybe we could support passing the class in directly instead of constructing an instance yourself. That would avoid the memo. I tried to deprecate passing options to the constructor because it invalidates too much. Maybe other custom layouts could use it for something though, not sure. We could probably support both ways.